### ISSUE TITLE
EnableDataBinding PropertyChanged PropertyChangedEventArgs.PropertyName always null

### ISSUE THREAD

**Issue by djwork** [mganss/XmlSchemaClassGenerator#522]
Hi mganss
           I am trying to use the PropertyChanged event handler but I seem to be getting null from PropertyChangedEventArgs.PropertyName.

I noticed that "propertyName = null" is being passed to the PropertyChangedEventArgs constructor inside the generated OnPropertyChanged method.

```
protected virtual void OnPropertyChanged([System.Runtime.CompilerServices.CallerMemberNameAttribute()] string propertyName = null)
{
    PropertyChanged?.Invoke(this, new System.ComponentModel.PropertyChangedEventArgs(propertyName = null));
}
```

I manually changed "PropertyChangedEventArgs(propertyName = null))" -> "PropertyChangedEventArgs(propertyName))" and the PropertyChangedEventArgs.PropertyName came through correctly.

I think this is a bug in ClassModel.Generate()

```
var param = new CodeParameterDeclarationExpression(typeof(string), "propertyName = null");
param.CustomAttributes.Add(new(TypeRef<System.Runtime.CompilerServices.CallerMemberNameAttribute>()));
var threadSafeDelegateInvokeExpression = new CodeSnippetExpression($"{propertyChangedEvent.Name}?.Invoke(this, new System.ComponentModel.PropertyChangedEventArgs({param.Name}))");

```
param's Name property is set to "propertyName = null" via the constructor, which is correct because CallerMemberNameAttribute requires the parameter to have a default value.
But the string passed to the CodeSnippetExpression constructor reuses the param.Name which is what I believe is creating the bug, because the Name property contains the parameter default expression:
![image](https://github.com/mganss/XmlSchemaClassGenerator/assets/6119701/59a1ab70-3b35-4d20-8bc9-5910d00d8c8c)

So what do think about changing:
```
var threadSafeDelegateInvokeExpression = new CodeSnippetExpression($"{propertyChangedEvent.Name}?.Invoke(this, new System.ComponentModel.PropertyChangedEventArgs({param.Name}))");
```
to:
```
var threadSafeDelegateInvokeExpression = new CodeSnippetExpression($"{propertyChangedEvent.Name}?.Invoke(this, new System.ComponentModel.PropertyChangedEventArgs(propertyName))");
```

**Comment by mganss** [comment]
Thanks for catching this one and also providing the fix!

**Comment by mganss** [comment]
Released in 2.1.1155.
